### DISCUSSION TITLE
srsUE function split b/w USRP and host SW

### DISCUSSION THREAD

**Discussion by abdyukin** [srsran/srsRAN_Project#473] [General Help]
Hello, srsRAN team!
In our project we want to debug our gNB PHY solution, using srsUE, deployed in USRP N310 setup environment. Our scenario however is 100MHz with 30kHz SCS, which is not supported yet by srsUE. Thus we want to mirror the demodulated IQ samples for further post-processing and debug to our proprietary SW.
May you please help to share the information about the srsUE processing, done on USRP side and on srsRAN host side?
Thanks in advance!

**Comment by pgawlowicz** [comment]
Hi @abdyukin, could you elaborate a bit more on what you want to achieve? As you said the srsUE does not support 100MHz BW with 30kHz SCS, so it will not connect to the gnb. You can capture the signal using another URSP and use it for further processing.

**Reply by abdyukin** [reply]
Hi @pgawlowicz , thanks for your respond!
Sorry to confuse you. Actually, i need to run a very basic test of our gNB L1 soultion, namely PDSCH with full allocation only. I am not targetting srsUE to complete the attach procedure, but i thought it might be possible to extract e.g. equalyzed DL IQ samples from somewhere in srsUE processing chain. I have N310 USRP and i assume it will do the ADC and DDC steps, while baseband signal processing is allocated for srsUE SW. If so, i may deepdive into the srsUE code to grab needed samples for e.g. QAM constellation analysis. What is your view on this? Thanks in advance!

**Reply by pgawlowicz** [reply]
Hi @abdyukin, srsUE supports max 20MHz bandwidth with 15kHz SCS. Therefore, it is not possible to get IQ samples of 100MHz signal from it. It will not even try to synchronize or demodulate the signal. On the other hand, you can simply capture the signal using N310, put it to file and use it for further processing. You will need to correctly synchronize to a slot boundary based on SSB.

**Reply by abdyukin** [reply]
Hi @pgawlowicz , thanks a lot for your suggestion and clarification. We will give it a try.
